src/test/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/WhitelistApiTest.java

```diff
@@ -0,0 +1,211 @@
+    }
+
+    /**
+     * Tests 4 scenarios for accessing and using the API.
+     * No creds, no admin certificate - UNAUTHORIZED
+     * non admin creds, no admin certificate - FORBIDDEN
+     * admin creds, no admin certificate - FORBIDDEN
+     * any creds, admin certificate - OK
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testWhitelistApi() throws Exception {
+        setupWithRestRoles(null);
+
+        {
+            // No creds, no admin certificate - UNAUTHORIZED
+            rh.keystore = "restapi/kirk-keystore.jks";
+            rh.sendAdminCertificate = false;
+            testGetAndPut(HttpStatus.SC_UNAUTHORIZED);
+        }
+
+
+        {
+            //non admin creds, no admin certificate - FORBIDDEN
+            rh.keystore = "restapi/kirk-keystore.jks";
+            rh.sendAdminCertificate = false;
+            testGetAndPut(HttpStatus.SC_FORBIDDEN, nonAdminCredsHeader);
+        }
+
+        {
+            // admin creds, no admin certificate - FORBIDDEN
+            rh.keystore = "restapi/kirk-keystore.jks";
+            rh.sendAdminCertificate = false;
+            testGetAndPut(HttpStatus.SC_FORBIDDEN, adminCredsHeader);
+        }
+
+        {
+            // any creds, admin certificate - OK
+            rh.keystore = "restapi/kirk-keystore.jks";
+            rh.sendAdminCertificate = true;
+            testGetAndPut(HttpStatus.SC_OK, nonAdminCredsHeader);
+        }
+    }
+
+    @Test
+    public void testWhitelistAuditComplianceLogging() throws Exception {
+        Settings settings = Settings.builder()
+                .put("opendistro_security.audit.type", TestAuditlogImpl.class.getName())
+                .put(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_ENABLE_TRANSPORT, false)
+                .put(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_ENABLE_REST, false)
+                .put(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_RESOLVE_BULK_REQUESTS, false)
+                .put(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_WRITE_LOG_DIFFS, true)
+                .put(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_EXTERNAL_CONFIG_ENABLED, false)
+                .put(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_INTERNAL_CONFIG_ENABLED, true)
+                .put(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DISABLED_TRANSPORT_CATEGORIES, "authenticated,GRANTED_PRIVILEGES")
+                .put(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DISABLED_REST_CATEGORIES, "authenticated,GRANTED_PRIVILEGES")
+                .build();
+        setupWithRestRoles(settings);
+        TestAuditlogImpl.clear();
+
+        {
+            // any creds, admin certificate - OK
+            rh.keystore = "restapi/kirk-keystore.jks";
+            rh.sendAdminCertificate = true;
+            testGetAndPut(HttpStatus.SC_OK, nonAdminCredsHeader);
+        }
+
+        System.out.println(TestAuditlogImpl.sb.toString());
```

Comment:
may want to remove any print statements used for debugging.

Reply:
Yes, I will. I copied this test case from NodesDnApiTest, which still has this println statement, and I forgot to remove it.

Reply:
Fixed